src/vehicle.cpp

```diff
@@ -4693,7 +4773,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+            if( chainsaw_on && part_info( ret.part ).has_flag( "CHAINSAW" ) &&
+                parts[part].hp > 0 ) {
+                dam += 66;
```

Comment:
@Coolthulhu is it what you wanted? If it is then I need some feedback about exact value here but if it should be done in another way (to respect target's armor) then I need a clearer clue how to do that.